libkbfs/disk_journal.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strconv"
+)
+
+// diskJournal stores an ordered list of entries.
+//
+// The directory layout looks like:
+//
+// dir/EARLIEST
+// dir/LATEST
+// dir/0...000
+// dir/0...001
+// dir/0...fff
+//
+// Each file in dir is named with an ordinal and contains a generic
+// serializable entry object. The files EARLIEST and LATEST point to
+// the earliest and latest valid ordinal, respectively.
+//
+// TODO: Do all high-level operations atomically on the file-system
```

Comment:
It looks like you're depending on higher-level synchronization to make sure no two entries get the same ordinal -- even use atomic file system operations won't help that.  That's fine, but can you please add a comment about that assumption?